### ISSUE TITLE
Debug toolbar not showing

### ISSUE THREAD

**Issue by charlesyuJY** [jazzband/django-debug-toolbar#1481]
Hi,

I followed the setup instruction on the official website, however, I can not see the toolbar when running it locally. I tried almost every solution on the internet, still can't get it to work. 

I can see the toolbar by viewing the page source, but it is not showing on the page.

Any advice would be helpful. Thank you so much!

**Comment by tim-schilling** [comment]
Hi @charlesyuJY, can you provide us with some more information about your setup? Version of django, using asgi vs wsgi, version of python. Additionally, can you check to see if there are any errors in the browser console tab as well as in the terminal where you have Django running.

**Comment by charlesyuJY** [comment]
Hi, thank you for replying.

My setup is :
Django                                        3.2.5
django-debug-toolbar               3.2.1

WSGI

Python 3.8.5 (Anaconda)

I checked again, there are no errors when running the server. The toolbar is not visible, but I can see that the toolbar exist when checking the page source.
![image](https://user-images.githubusercontent.com/45712833/124630345-ead15900-deb4-11eb-8549-fd178f89cc05.png)
![image](https://user-images.githubusercontent.com/45712833/124630619-3126b800-deb5-11eb-8b77-430ef3420b69.png)

**Comment by tim-schilling** [comment]
Can you verify that your `HKEY_CLASSES_ROOT` registry key is setup correctly? You're specifically looking for the content type for JS. Here's a [comment discussing a fellow Windows developer's setup](https://github.com/jazzband/django-debug-toolbar/issues/1420#issuecomment-749331855).

**Comment by dahua910** [comment]
I am mac user, got same issue, Debug toolbar not showing
any help?

Django==3.2.5
django-debug-toolbar==3.2.1
pyenv Python 3.8.10

settings.py-----------------------------------
INSTALLED_APPS += [
    'debug_toolbar',
]
INTERNAL_IPS = ['127.0.0.1', ]
DEBUG = True
MIDDLEWARE= ['debug_toolbar.middleware.DebugToolbarMiddleware',]

url.py-----------------------------------
urlpatterns += [
    path('__debug__/', include(debug_toolbar.urls)),
]

**Comment by tim-schilling** [comment]
@dahua910 I can try to help. Can you please answer the questions I posed [here](https://github.com/jazzband/django-debug-toolbar/issues/1481#issuecomment-874869243)?

**Comment by AlexChapman18** [comment]
Hi, i am a windows 10 user currently experiencing the same issue.

Django==3.2.6
django-debug-toolbar==3.2.1 (Also tried with 3.2 with the same issue)
Python version  3.8.10

settings.py-----------------------------------
INSTALLED_APPS += [
'debug_toolbar',
]
INTERNAL_IPS = ['127.0.0.1', ]
DEBUG = True
MIDDLEWARE= ['debug_toolbar.middleware.DebugToolbarMiddleware',]

url.py-----------------------------------
import debug_toolbar
from django.contrib import admin
from django.urls import path, include


urlpatterns = [
    path('admin/', admin.site.urls),
    path("playground/", include("playground.urls")),
    path("__debug__/", include(debug_toolbar.urls)),
]

When attempting to load the toolbar onto the page i recive this error
"Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain""
I have followed advice from previous threads to change the registry key to "text/javascript" but this did not help the issue.

In the meantime i have switched to toolbar version 2.2 which seems to be working

If you need any more information just ask and i will reply as soon as i can

Thanks

**Comment by tim-schilling** [comment]
@AlexChapman18 did that resolve the error `Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain"`?

**Comment by AlexChapman18** [comment]
@tim-schilling Yes, changing to django-debug-toolbar==2.2 did resolve the error

**Comment by tim-schilling** [comment]
I should have been more clear. Did changing your registry mime type for `.js` files to JavaScript as indicated in the [troubleshooting part of the docs](https://django-debug-toolbar.readthedocs.io/en/latest/installation.html#troubleshooting) resolve the error `Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain"`?

**Comment by AlexChapman18** [comment]
no, from what i could tell changing the registry key had no effect on toolbar version 3.2.1 or 3.2

**Comment by tim-schilling** [comment]
If you're still getting that error message, then I think the change to the registry wasn't correct see this SO answer for correct formatting: https://stackoverflow.com/a/64055514/1637351

**Comment by tim-schilling** [comment]
@AlexChapman18 were you able to resolve this issue?

**Comment by tim-schilling** [comment]
Closing due to lack of response.

**Comment by Prashant4900** [comment]
I also face the same problem that debug_toolbar not show on the webpage.

### **Step to reproduce**
- Setup virtual env with virtualenv venv.
- Create Django project v4.1.
- Run pip install django-debug-toolbar==3.5.0.
- Update settings.

### Urls.py

    from django.contrib import admin
    from django.urls import path, include
    
    urlpatterns = [
        path('admin/', admin.site.urls),
        path('__debug__/', include('debug_toolbar.urls')),
    ]

### Settings.py

    DEBUG = True
    
    ALLOWED_HOSTS = [
        '*'
    ]

    INSTALLED_APPS = [
        'django.contrib.admin',
        'django.contrib.auth',
        'django.contrib.contenttypes',
        'django.contrib.sessions',
        'django.contrib.messages',
        'django.contrib.staticfiles',
        'debug_toolbar',
    ]

    MIDDLEWARE = [
        'django.middleware.security.SecurityMiddleware',
        'django.contrib.sessions.middleware.SessionMiddleware',
        'django.middleware.common.CommonMiddleware',
        'django.middleware.csrf.CsrfViewMiddleware',
        'django.contrib.auth.middleware.AuthenticationMiddleware',
        'django.contrib.messages.middleware.MessageMiddleware',
        'django.middleware.clickjacking.XFrameOptionsMiddleware',
        'debug_toolbar.middleware.DebugToolbarMiddleware',
    ]

    TEMPLATES = [
        {
            'BACKEND': 'django.template.backends.django.DjangoTemplates',
            'DIRS': [],
            'APP_DIRS': True,
            'OPTIONS': {
                'context_processors': [
                    'django.template.context_processors.debug',
                    'django.template.context_processors.request',
                    'django.contrib.auth.context_processors.auth',
                    'django.contrib.messages.context_processors.messages',
                ],
            },
        },
    ]

    INTERNAL_IPS = [
        '127.0.0.1',
        'localhost',
    ]
    
    STATIC_URL = 'static/'


### Preview

![image](https://user-images.githubusercontent.com/62440911/184505129-885d0440-6a38-47ff-8386-31f93f4b94b3.png)
![image](https://user-images.githubusercontent.com/62440911/184505253-e81d110a-95b3-4f0f-b74b-7691d44cba95.png)
![image](https://user-images.githubusercontent.com/62440911/184505296-01f805d1-b06f-42ce-b01b-e38cb2bec6e2.png)

**Comment by tombohub** [comment]
I have toolbar showing on index page, but not on the other page

**Comment by vitorgomateus** [comment]
I just installed django-debug-toolbar, and followed the installation guide, but the toolbar is not showing. I tried downgrading to the 2.2 version, but got a couple of errors, so I came back to the latest one. I get no error in the command line or console, so I am assuming it is not related with the MIME type issue. I also tried doing this but nothing happens:
```python
def show_toolbar(request):
    return True
DEBUG_TOOLBAR_CONFIG = {
    "SHOW_TOOLBAR_CALLBACK" : show_toolbar,
}
``` 

Name: django-debug-toolbar
Version: 4.1.0
---
Django version 4.2.2

Any help is appreciated.

**Comment by maisonhai3** [comment]
I got the same problem in django 4.2.6.

**Comment by maisonhai3** [comment]
@tim-schilling 
I figured it out. 

### The problem
The problem did not come from the debug toolbar but from the toolbar installation guide.

In my case, because the Django template has no HTML `<body>` tag, the debug toolbar didn't appear.
Using an admin template instead, which has a proper HTML `<body>` and the toolbar perfectly shows up.
However, the installation guide did not mention the need for the `<body>`.

### The scenario
1. I was working on the Django tutorial, [part 8](https://docs.djangoproject.com/en/4.2/intro/tutorial08/), which had me install the debug toolbar.
2. I precisely followed the debug toolbar [installation guide](https://django-debug-toolbar.readthedocs.io/en/latest/installation.html).
3. However, the toolbar did not appear on my simple homepage built with the Django tutorial.
![image](https://github.com/jazzband/django-debug-toolbar/assets/28621061/9329afeb-943a-4057-9d22-c17eb544b98b)

5. After searching, I tried suggestions in the comments above. Nothing worked.
6. Then, Bing AI said that maybe the page didn't include a `<body>` tag. (This is not an ad :smile: ). That assumption was confirmed when the admin page had the loveable toolbar.

![image](https://github.com/jazzband/django-debug-toolbar/assets/28621061/8c8b674a-ddf8-46a9-86d5-e365c42c3f94)

Simply put, please somehow put a word of `<body>` tag in the docs.

**Comment by tim-schilling** [comment]
I'm glad you figured it out. It seems like the project could do a better job of helping a person when they're stuck.

> Simply put, please somehow put a word of <body> tag in the docs.

It's in the docs under the section ["The toolbar isn't displayed!"](https://django-debug-toolbar.readthedocs.io/en/latest/tips.html#the-toolbar-isn-t-displayed)

> It will also only display if the MIME type of the response is either text/html or application/xhtml+xml and **contains a closing </body> tag**.

**Comment by JonneSaloranta** [comment]
Thanks @vitorgomateus for the code snippet. It fixed my issue.
No idea what is different since I've never had this issue before.

```python
def show_toolbar(request):
    return True
DEBUG_TOOLBAR_CONFIG = {
    "SHOW_TOOLBAR_CALLBACK" : show_toolbar,
}
```

django-debug-toolbar 4.4.6
Django version 4.2.16
